### PR TITLE
No op for highlighting and symbol references in issues mode

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/source/DefaultHighlightable.java [hunk 1]
```diff
@@ -32,15 +33,17 @@
...
-  public DefaultHighlightable(DefaultInputFile inputFile, SensorStorage sensorStorage) {
+  public DefaultHighlightable(DefaultInputFile inputFile, SensorStorage sensorStorage, AnalysisMode analysisMode) {
     this.inputFile = inputFile;
     this.sensorStorage = sensorStorage;
+    this.analysisMode = analysisMode;
   }
 
   @Override
   public HighlightingBuilder newHighlighting() {
-    DefaultHighlighting defaultHighlighting = new DefaultHighlighting(sensorStorage);
```
Comment (dbmeneses) [inline sonar-batch/src/main/java/org/sonar/batch/source/DefaultHighlightable.java]: Wouldn't it be better if the builder builds a no-op object (NoOpHighlighting), leaving DefaultHighlighting unchanged?

**Comment by dbmeneses** [comment]
LGTM appart of the comment, that also applies to SymbolTable.

**Comment by henryju** [comment]
@dbmeneses Good idea, please have a look at new implementation.

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**